server/app/files.go

```diff
@@ -141,7 +141,8 @@ func (a *App) GetFilePath(teamID, boardID, fileName string) (*mm_model.FileInfo,
 
 func getDestinationFilePath(isTemplate bool, teamID, boardID, filename string) (string, error) {
 	// Validate inputs to ensure proper file path handling
-	if !mm_model.IsValidId(teamID) {
+	// Only allow GlobalTeamID for template operations to prevent path traversal attacks
+	if !mm_model.IsValidId(teamID) && !(isTemplate && teamID == model.GlobalTeamID) {
```

Comment:
Whenever we will validate team ID, we will have to handle the global team ID case for templates. So, it would be better if we create a function that encapsulates the whole logic of validating team IDs, something like `ValidateTeamId(teamId, isTemplate)`.